Tests/Web5Tests/Crypto/JOSE/JWTTests.swift

```diff
@@ -21,6 +22,133 @@ final class JWTTests: XCTestCase {
+            XCTAssertEqual(error, JWT.Error.verificationFailed("Expected JWT header to contain kid"))
+        }
+
+        // decode payload fails
+        let header = JWS.Header(algorithm: .es256k, keyID: "keyID", type: "JWT")
+        let base64UrlEncodedHeader = try JSONEncoder().encode(header).base64UrlEncodedString()
+        XCTAssertThrowsError(try JWT.parse(jwtString: "\(base64UrlEncodedHeader).efgh.hijk"))
+    }
+
+    func testVerifyJwtExpired() async throws {
+        let did = try DIDJWK.create(options: .init(algorithm: .secp256k1))
+        let header = JWS.Header(algorithm: .es256k, keyID: did.document.verificationMethod!.first!.id, type: "JWT")
+        let base64UrlEncodedHeader = try JSONEncoder().encode(header).base64UrlEncodedString()
+        let claims = JWT.Claims(expiration: Date(timeIntervalSinceNow: -1))
+        let base64UrlEncodedClaims = try JSONEncoder().encode(claims).base64UrlEncodedString()
+        do {
+             _ = try await JWT.verify(jwt: "\(base64UrlEncodedHeader).\(base64UrlEncodedClaims).hijk")
+        } catch let error as JWT.Error {
+            XCTAssertEqual(error, JWT.Error.verificationFailed("JWT has expired"))
+        }
+    }
+
+    func testVerifyJwtKeyIdNotDereferenceVerificationMethod() async throws {
+        let did = try DIDJWK.create(options: .init(algorithm: .secp256k1))
+        let header = JWS.Header(algorithm: .es256k, keyID: did.uri, type: "JWT")
+        let base64UrlEncodedHeader = try JSONEncoder().encode(header).base64UrlEncodedString()
+        let claims = JWT.Claims(issuer: did.uri, subject: did.uri, issuedAt: Date())
+        let base64UrlEncodedClaims = try JSONEncoder().encode(claims).base64UrlEncodedString()
+        do {
+             _ = try await JWT.verify(jwt: "\(base64UrlEncodedHeader).\(base64UrlEncodedClaims).hijk")
+        } catch let error as JWT.Error {
+            XCTAssertEqual(error, JWT.Error.verificationFailed("Expected kid in JWT header to dereference a DID Document Verification Method"))
+        }
+    }
+
+    func testVerifyPublicKeyAlgorithmIsNotSupported() async throws {
+        let did = try DIDJWK.create(options: .init(algorithm: .secp256k1))
+        let header = JWS.Header(algorithm: .eddsa, keyID: did.document.verificationMethod!.first!.id, type: "JWT")
+        let base64UrlEncodedHeader = try JSONEncoder().encode(header).base64UrlEncodedString()
+        let claims = JWT.Claims(issuer: did.uri, subject: did.uri, issuedAt: Date())
+        let base64UrlEncodedClaims = try JSONEncoder().encode(claims).base64UrlEncodedString()
+        do {
+             _ = try await JWT.verify(jwt: "\(base64UrlEncodedHeader).\(base64UrlEncodedClaims).hijk")
+        } catch let error as JWT.Error {
+            XCTAssertEqual(error, JWT.Error.verificationFailed("Expected alg in JWT header to match DID Document Verification Method alg"))
+        }
+    }
+
+    func testReturnSignerDidIfVerificationSucceeds() async throws {
+        let didStr = """
```

Comment:
We probably don't need to import a portable DID here. Could we `let did = try DIDJWK.create()` to ensure we are only testing the success of `verify` here?

Reply:
Yes you are right. we don't need to import a portable DID here.